tendermint/src/rpc/client.rs

```diff
@@ -182,10 +183,44 @@ impl Client {
                     .unwrap(),
             );
         }
-
+        dbg!(&request);
         let http_client = hyper::Client::builder().build_http();
         let response = http_client.request(request).await?;
         let response_body = hyper::body::aggregate(response.into_body()).await?;
         R::Response::from_reader(response_body.reader())
     }
+
+    //TODO Have a query type instead of a string
+    /// Subscribe to the Events Websocket with a query string for example "tm.event = 'NewBlock'"
+    pub async fn subscribe(&mut self, query: &str) -> Result<(), Box<dyn std::error::Error>> {
+        let (host, port) = match &self.address {
+            net::Address::Tcp { host, port, .. } => (host, port),
+            other => {
+                let err: Box<dyn std::error::Error> =
+                    Error::invalid_params(&format!("invalid RPC address: {:?}", other)).into();
+
+                return Err(err);
+            }
+        };
+        let mut ws =
+            subscribe::WebSocketEvents::websocket(&format!("ws://{}:{}", host, port)).await?;
+        ws.subscribe(query).await?;
+
+        self.ws = Some(ws);
+
+        Ok(())
+    }
+
+    //TODO Have a query type instead of a string
+    /// Subscribe to the Events Websocket with a query string for example "tm.event = 'NewBlock'"
+    pub async fn get_event(&mut self) -> Result<subscribe::Event, Box<dyn std::error::Error>> {
```

Comment:
Maybe we can remove the `Box<dyn std::error::Error>` altogether by implementing `impl From<Box<dyn std::error::Error>> for rpc::Error`. Then this method can return `rpc::Error` instead of `Box<dyn std::error::Error>` (consistent with the other methods).